src/date.rs

```diff
@@ -43,10 +43,24 @@ pub struct Date {
 
 impl fmt::Debug for Date {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
-        f.debug_struct("Date")
-            .field("year", &self.year())
-            .field("ordinal", &self.ordinal())
-            .finish()
+        if cfg!(feature = "large-dates") && self.year().abs() >= 10_000 {
+            write!(
+                f,
+                "{:+}-{:02}-{:02}",
+                self.year(),
+                self.month() as u8,
+                self.day()
+            )
+        } else {
+            write!(
+                f,
+                "{:0width$}-{:02}-{:02}",
+                self.year(),
+                self.month() as u8,
+                self.day(),
+                width = 4 + (self.year() < 0) as usize
+            )
+        }
```

Comment:
Let's reuse the existing `Display` implementation. No need to duplicate code, after all.

```suggestion
        fmt::Display::fmt(self, f)
```